src/service/performance-impl.js

```diff
@@ -421,6 +436,8 @@ export class Performance {
     }
 
     if (entryTypesToObserve.length === 0) {
+      // No entries to observe, so resolve promise
+      this.fcpResolve_();
```

Comment:
I'm not sure about this. It is possible that this is because `first_contentful_paint` hasn't arrived yet. I don't think we should resolve here. 

@jridgewell We want to resolve the fcp promise if it fails to arrive. Do you know what's the expected behavior of performance Observer, is the 'fcp' entry guaranteed to arrive? Thanks.

Reply:
🤷

Reply:
Discussed offline: this is not needed if we use the solution that checks for the platform.